### PR TITLE
Update branding to 3.0.0-preview

### PR DESCRIPTION
**Pull request by natemcmaster**
This _mostly_ implements https://github.com/dotnet/arcade/blob/master/Documentation/CorePackages/Versioning.md.

Changes:
* This uses date-based versioning
* Builds are `3.0.0-preview-$(buildnumber)`, not alpha1 or preview1.

Notable deviations:

* This uses a 4x left-padded number, not 2x. This is temporary until we get official builds running in Azure Pipelines. 4x is necessary because TeamCity build counters do not reset each day.
* This uses semver 1.0, not 2.0 like the Extensions or EFCore packages. We can update to semver 2.0 if https://github.com/dotnet/core-setup/issues/4795 is resolved.

### CODE REVIEW

**Comment by natemcmaster** [comment]
Test failures appear to be unrelated to this change: https://github.com/aspnet/AspNetCore-Internal/issues/1395

**Comment by natemcmaster** [comment]
FYI @jkotalik for review of the impact on ANCM versioning.

File: version.props [hunk 1]
```diff
@@ -3,13 +3,43 @@
...
-    <PreReleaseLabel>alpha1</PreReleaseLabel>
-    <PreReleaseBrandingLabel>Alpha 1</PreReleaseBrandingLabel>
-    <BuildNumber Condition="'$(BuildNumber)' == '' OR '$(UsingLocalBuildNumber)' == 'true'">$([System.DateTime]::Now.ToString('yyMMdd'))-99</BuildNumber>
-    <VersionPrefix>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).$(AspNetCorePatchVersion)</VersionPrefix>
+    <PreReleaseLabel>preview</PreReleaseLabel>
+    <PreReleaseBrandingLabel>Preview</PreReleaseBrandingLabel>
     <ExperimentalVersionPrefix>0.3.$(AspNetCorePatchVersion)</ExperimentalVersionPrefix>
-    <VersionSuffix>$(PreReleaseLabel)-$(BuildNumber)</VersionSuffix>
-    <BrandingVersionSuffix>$(PreReleaseBrandingLabel) Build $(BuildNumber)</BrandingVersionSuffix>
+    <AspNetCoreModuleVersionMajor>1$(AspNetCoreMajorVersion)</AspNetCoreModuleVersionMajor>
```
Comment (Eilon) [inline version.props]: Is the `1` here on purpose? Is that what the question for @jkotalik is?

Comment (natemcmaster) [reply version.props]: I added a comment to clarify why this is done. It is intentional.

**Review by Eilon — APPROVED** [review]
Looks good to me. Also added @shirhatti because of the effect on ANCM versioning.

**Pull request merged**